### PR TITLE
`requirements.txt`: Pin `onnxruntime>=1.7.0` rather than `==1.4.0`

### LINKED ISSUE #3433: Unpin `onnxruntime` due to OpenMP requirement being removed in v1.7.0
We're currently pinning `onnxruntime==1.4.0` to avoid the OpenMP dependency from `onnxruntime>=1.5.1`. (https://github.com/spinalcordtoolbox/spinalcordtoolbox/issues/3033)

However, [`onnxuntime==1.7.0`](https://github.com/microsoft/onnxruntime/releases/tag/v1.7.0) removed this requirement, so we no longer have to pin.

This was one of the blockers for Python 3.9 support (https://github.com/spinalcordtoolbox/spinalcordtoolbox/issues/3367), so huzzah!

### PR DESCRIPTION
**Pull request by joshuacwnewton**
#### GitHub

- [x] I've given this PR a concise, self-descriptive, and meaningful title
- [x] I've linked relevant issues in the PR body
- [x] I've applied [the relevant labels](https://www.neuro.polymtl.ca/software/contributing#pr_labels) to this PR
- [x] I've applied a [release milestone](https://github.com/spinalcordtoolbox/spinalcordtoolbox/milestones) (major, minor, patch) in line with [Semantic Versioning guidelines](https://github.com/spinalcordtoolbox/spinalcordtoolbox/wiki/Misc%3A-Creating-a-new-release#convention-for-naming-releases) 
- [x] I've assigned a reviewer

#### PR contents

- [ ] I've consulted [SCT's internal developer documentation](https://github.com/spinalcordtoolbox/spinalcordtoolbox/wiki) to ensure my contribution is in line with any relevant design decisions
- [ ] I've added [relevant tests](https://github.com/spinalcordtoolbox/spinalcordtoolbox/wiki/Programming%3A-Tests) for my contribution
- [ ] I've updated the [relevant documentation](https://github.com/spinalcordtoolbox/spinalcordtoolbox/wiki/Programming%3A-Documentation) for my changes, including argparse descriptions, docstrings, and ReadTheDocs tutorial pages

## Description

Previously, we pinned `onnxruntime==1.4.0` because of a dependency on OpenMP on macOS, which required a cumbersome `brew install openmp` step we didn't want to impose on our users.

For versions of `onnxruntime>=1.7.0`, however, this requirement has been removed. Ivadomed has already updated, and now we can do the same.

## Linked issues

Fixes #3433.

### CODE REVIEW

File: requirements.txt
Comment (joshuacwnewton) [inline requirements.txt:14]: It looks like moving away from `onnxruntime==1.4.0` has the side-effect of... _downgrading_ us from `ivadomed=2.7.4` to `ivadomed=2.4.0`? 

Digging deeper...

Comment (joshuacwnewton) [reply requirements.txt:14]: Ahh, okay. The reason for this is partially due to https://github.com/ivadomed/ivadomed/issues/1042. We can't access the newer versions of `ivadomed` due to a dependency conflict re: `tensorboard`.

**Review by jcohenadad — COMMENTED** [review]
I'm currently testing with 
~~~
git+https://github.com/ivadomed/ivadomed@master
~~~

I'll let you know how it goes

**Comment by joshuacwnewton** [comment]
Thanks, @jcohenadad! I think we'll be blocked on this PR until https://github.com/ivadomed/ivadomed/issues/1042 is addressed, though.

**Comment by jcohenadad** [comment]
hum, i got an error:

<details><summary>Terminal output</summary>

```console
Installing Python dependencies...


Using requirements.txt (git installation)

Looking in links: https://download.pytorch.org/whl/cpu/torch_stable.html
Collecting git+https://github.com/ivadomed/ivadomed@master (from -r requirements.txt (line 7))
  Cloning https://github.com/ivadomed/ivadomed (to revision master) to /private/var/folders/s8/4qnm5q1n261ch35b5kkclsb00000gn/T/pip-req-build-s0sdca3h
  Running command git clone --filter=blob:none -q https://github.com/ivadomed/ivadomed /private/var/folders/s8/4qnm5q1n261ch35b5kkclsb00000gn/T/pip-req-build-s0sdca3h
  Resolved https://github.com/ivadomed/ivadomed to commit 0fdc5a8f7f3a93a86aa94ff3feac3c460c732ff5
  Preparing metadata (setup.py): started
  Preparing metadata (setup.py): finished with status 'done'
Ignoring torch: markers 'sys_platform != "darwin"' don't match your environment
Ignoring torchvision: markers 'sys_platform != "darwin"' don't match your environment
Collecting colored
  Using cached colored-1.4.3-py3-none-any.whl
Collecting dipy
  Using cached dipy-1.4.1-cp37-cp37m-macosx_10_9_x86_64.whl (7.6 MB)
Collecting h5py~=2.10.0
  Using cached h5py-2.10.0-cp37-cp37m-macosx_10_6_intel.whl (3.0 MB)
Collecting Keras==2.3.1
  Using cached Keras-2.3.1-py2.py3-none-any.whl (377 kB)
Collecting matplotlib
  Using cached matplotlib-3.5.1-cp37-cp37m-macosx_10_9_x86_64.whl (7.3 MB)
Collecting nibabel
  Using cached nibabel-3.2.1-py3-none-any.whl (3.3 MB)
Collecting numpy
  Using cached numpy-1.21.5-cp37-cp37m-macosx_10_9_x86_64.whl (16.9 MB)
Collecting onnxruntime>=1.7.0
  Downloading onnxruntime-1.10.0-cp37-cp37m-macosx_10_14_x86_64.whl (5.4 MB)
Collecting pandas
  Using cached pandas-1.3.5-cp37-cp37m-macosx_10_9_x86_64.whl (11.0 MB)
Collecting psutil
  Using cached psutil-5.9.0-cp37-cp37m-macosx_10_9_x86_64.whl (238 kB)
Collecting pyqt5==5.11.3
  Using cached PyQt5-5.11.3-5.11.2_a-cp35.cp36.cp37.cp38-abi3-macosx_10_6_intel.whl (91.1 MB)
Collecting pytest
  Using cached pytest-6.2.5-py3-none-any.whl (280 kB)
Collecting pytest-cov
  Using cached pytest_cov-3.0.0-py3-none-any.whl (20 kB)
Collecting raven
  Using cached raven-6.10.0-py2.py3-none-any.whl (284 kB)
Collecting requests
  Using cached requests-2.27.1-py2.py3-none-any.whl (63 kB)
Collecting requirements-parser
  Using cached requirements_parser-0.4.0-py3-none-any.whl (16 kB)
Collecting scipy
  Using cached scipy-1.7.3-cp37-cp37m-macosx_10_9_x86_64.whl (33.0 MB)
Collecting scikit-image
  Using cached scikit_image-0.19.1-cp37-cp37m-macosx_10_13_x86_64.whl (13.1 MB)
Collecting scikit-learn
  Using cached scikit_learn-1.0.2-cp37-cp37m-macosx_10_13_x86_64.whl (7.8 MB)
Collecting tensorflow~=1.15.0
  Using cached tensorflow-1.15.5-cp37-cp37m-macosx_10_9_x86_64.whl (123.3 MB)
Collecting torch==1.5.0
  Using cached torch-1.5.0-cp37-none-macosx_10_9_x86_64.whl (80.5 MB)
Collecting torchvision==0.6.0
  Using cached torchvision-0.6.0-cp37-cp37m-macosx_10_9_x86_64.whl (436 kB)
Collecting xlwt
  Using cached xlwt-1.3.0-py2.py3-none-any.whl (99 kB)
Collecting tqdm
  Using cached tqdm-4.62.3-py2.py3-none-any.whl (76 kB)
Collecting transforms3d
  Using cached transforms3d-0.3.1-py3-none-any.whl
Collecting urllib3[secure]
  Downloading urllib3-1.26.8-py2.py3-none-any.whl (138 kB)
Collecting pytest_console_scripts
  Downloading pytest-console-scripts-1.2.2.tar.gz (14 kB)
  Preparing metadata (setup.py): started
  Preparing metadata (setup.py): finished with status 'done'
Collecting wquantiles
  Using cached wquantiles-0.6-py3-none-any.whl (3.3 kB)
Collecting pyyaml
  Using cached PyYAML-6.0-cp37-cp37m-macosx_10_9_x86_64.whl (189 kB)
Collecting keras-applications>=1.0.6
  Using cached Keras_Applications-1.0.8-py3-none-any.whl (50 kB)
Collecting six>=1.9.0
  Using cached six-1.16.0-py2.py3-none-any.whl (11 kB)
Collecting keras-preprocessing>=1.0.5
  Using cached Keras_Preprocessing-1.1.2-py2.py3-none-any.whl (42 kB)
Collecting PyQt5_sip<4.20,>=4.19.11
  Using cached PyQt5_sip-4.19.19-cp37-cp37m-macosx_10_6_intel.whl (52 kB)
Collecting future
  Using cached future-0.18.2-py3-none-any.whl
Collecting pillow>=4.1.1
  Downloading Pillow-9.0.0-cp37-cp37m-macosx_10_10_x86_64.whl (3.0 MB)
Collecting pyparsing<3,>=2.0.2
  Using cached pyparsing-2.4.7-py2.py3-none-any.whl (67 kB)
Collecting csv-diff>=1.0
  Using cached csv_diff-1.1-py3-none-any.whl (12 kB)
Collecting loguru~=0.5
  Using cached loguru-0.5.3-py3-none-any.whl (57 kB)
Collecting joblib~=1.0
  Using cached joblib-1.1.0-py2.py3-none-any.whl (306 kB)
Collecting onnxruntime>=1.7.0
  Using cached onnxruntime-1.7.0-cp37-cp37m-macosx_10_12_x86_64.whl (4.6 MB)
Collecting pybids>=0.12.4
  Using cached pybids-0.14.0-py3-none-any.whl (3.2 MB)
Collecting seaborn~=0.11
  Using cached seaborn-0.11.2-py3-none-any.whl (292 kB)
Collecting tensorboard~=2.4
  Using cached tensorboard-2.7.0-py3-none-any.whl (5.8 MB)
Collecting torchio>=0.18.68
  Using cached torchio-0.18.71-py2.py3-none-any.whl (164 kB)
Collecting protobuf
  Downloading protobuf-3.19.3-cp37-cp37m-macosx_10_9_x86_64.whl (1.0 MB)
Collecting fonttools>=4.22.0
  Using cached fonttools-4.28.5-py3-none-any.whl (890 kB)
Collecting cycler>=0.10
  Using cached cycler-0.11.0-py3-none-any.whl (6.4 kB)
Collecting kiwisolver>=1.0.1
  Using cached kiwisolver-1.3.2-cp37-cp37m-macosx_10_9_x86_64.whl (61 kB)
Collecting python-dateutil>=2.7
  Using cached python_dateutil-2.8.2-py2.py3-none-any.whl (247 kB)
Collecting packaging>=20.0
  Using cached packaging-21.3-py3-none-any.whl (40 kB)
Collecting pytz>=2017.3
  Using cached pytz-2021.3-py2.py3-none-any.whl (503 kB)
Collecting toml
  Using cached toml-0.10.2-py2.py3-none-any.whl (16 kB)
Collecting importlib-metadata>=0.12
  Downloading importlib_metadata-4.10.1-py3-none-any.whl (17 kB)
Collecting py>=1.8.2
  Using cached py-1.11.0-py2.py3-none-any.whl (98 kB)
Collecting iniconfig
  Using cached iniconfig-1.1.1-py2.py3-none-any.whl (5.0 kB)
Collecting attrs>=19.2.0
  Downloading attrs-21.4.0-py2.py3-none-any.whl (60 kB)
Collecting pluggy<2.0,>=0.12
  Using cached pluggy-1.0.0-py2.py3-none-any.whl (13 kB)
Collecting coverage[toml]>=5.2.1
  Using cached coverage-6.2-cp37-cp37m-macosx_10_9_x86_64.whl (179 kB)
Requirement already satisfied: certifi>=2017.4.17 in ./python/envs/venv_sct/lib/python3.7/site-packages (from requests->-r requirements.txt (line 23)) (2021.10.8)
Collecting idna<4,>=2.5
  Using cached idna-3.3-py3-none-any.whl (61 kB)
Collecting charset-normalizer~=2.0.0
  Downloading charset_normalizer-2.0.10-py3-none-any.whl (39 kB)
Collecting types-setuptools>=57.0.0
  Downloading types_setuptools-57.4.7-py3-none-any.whl (27 kB)
Collecting networkx>=2.2
  Using cached networkx-2.6.3-py3-none-any.whl (1.9 MB)
Collecting tifffile>=2019.7.26
  Using cached tifffile-2021.11.2-py3-none-any.whl (178 kB)
Collecting imageio>=2.4.1
  Downloading imageio-2.13.5-py3-none-any.whl (3.3 MB)
Collecting PyWavelets>=1.1.1
  Using cached PyWavelets-1.2.0-cp37-cp37m-macosx_10_13_x86_64.whl (4.3 MB)
Collecting threadpoolctl>=2.0.0
  Using cached threadpoolctl-3.0.0-py3-none-any.whl (14 kB)
Requirement already satisfied: wheel>=0.26 in ./python/envs/venv_sct/lib/python3.7/site-packages (from tensorflow~=1.15.0->-r requirements.txt (line 28)) (0.37.1)
Collecting google-pasta>=0.1.6
  Using cached google_pasta-0.2.0-py3-none-any.whl (57 kB)
Collecting absl-py>=0.7.0
  Using cached absl_py-1.0.0-py3-none-any.whl (126 kB)
Collecting opt-einsum>=2.3.2
  Using cached opt_einsum-3.3.0-py3-none-any.whl (65 kB)
Collecting numpy
  Using cached numpy-1.18.5-cp37-cp37m-macosx_10_9_x86_64.whl (15.1 MB)
Collecting tensorflow-estimator==1.15.1
  Using cached tensorflow_estimator-1.15.1-py2.py3-none-any.whl (503 kB)
Collecting astor>=0.6.0
  Using cached astor-0.8.1-py2.py3-none-any.whl (27 kB)
Collecting tensorflow~=1.15.0
  Using cached tensorflow-1.15.4-cp37-cp37m-macosx_10_9_x86_64.whl (123.3 MB)
  Using cached tensorflow-1.15.3-cp37-cp37m-macosx_10_11_x86_64.whl (123.4 MB)
  Using cached tensorflow-1.15.2-cp37-cp37m-macosx_10_11_x86_64.whl (123.3 MB)
  Using cached tensorflow-1.15.0-cp37-cp37m-macosx_10_11_x86_64.whl (124.0 MB)
INFO: pip is looking at multiple versions of scikit-learn to determine which version is compatible with other requirements. This could take a while.
Collecting scikit-learn
  Using cached scikit_learn-1.0.1-cp37-cp37m-macosx_10_13_x86_64.whl (7.9 MB)
INFO: pip is looking at multiple versions of scikit-image to determine which version is compatible with other requirements. This could take a while.
Collecting scikit-image
  Using cached scikit_image-0.19.0-cp37-cp37m-macosx_10_13_x86_64.whl (35.8 MB)
INFO: pip is looking at multiple versions of scipy to determine which version is compatible with other requirements. This could take a while.
Collecting scipy
  Using cached scipy-1.7.2-cp37-cp37m-macosx_10_9_x86_64.whl (33.0 MB)
INFO: pip is looking at multiple versions of requirements-parser to determine which version is compatible with other requirements. This could take a while.
Collecting requirements-parser
  Using cached requirements_parser-0.3.1-py3-none-any.whl (15 kB)
INFO: pip is looking at multiple versions of requests to determine which version is compatible with other requirements. This could take a while.
Collecting requests
  Downloading requests-2.27.0-py2.py3-none-any.whl (63 kB)
INFO: pip is looking at multiple versions of raven to determine which version is compatible with other requirements. This could take a while.
Collecting raven
  Using cached raven-6.9.0-py2.py3-none-any.whl (287 kB)
INFO: pip is looking at multiple versions of pytest-cov to determine which version is compatible with other requirements. This could take a while.
Collecting pytest-cov
  Using cached pytest_cov-2.12.1-py2.py3-none-any.whl (20 kB)
INFO: pip is looking at multiple versions of pytest to determine which version is compatible with other requirements. This could take a while.
Collecting pytest
  Using cached pytest-6.2.4-py3-none-any.whl (280 kB)
INFO: pip is looking at multiple versions of psutil to determine which version is compatible with other requirements. This could take a while.
Collecting psutil
  Using cached psutil-5.8.0-cp37-cp37m-macosx_10_9_x86_64.whl (236 kB)
INFO: pip is looking at multiple versions of pandas to determine which version is compatible with other requirements. This could take a while.
Collecting pandas
  Using cached pandas-1.3.4-cp37-cp37m-macosx_10_9_x86_64.whl (11.3 MB)
INFO: pip is looking at multiple versions of numpy to determine which version is compatible with other requirements. This could take a while.
Collecting numpy
  Using cached numpy-1.21.4-cp37-cp37m-macosx_10_9_x86_64.whl (16.9 MB)
INFO: pip is looking at multiple versions of nibabel to determine which version is compatible with other requirements. This could take a while.
Collecting nibabel
  Using cached nibabel-3.2.0-py3-none-any.whl (3.3 MB)
INFO: pip is looking at multiple versions of matplotlib to determine which version is compatible with other requirements. This could take a while.
Collecting matplotlib
  Using cached matplotlib-3.5.0-cp37-cp37m-macosx_10_9_x86_64.whl (7.3 MB)
INFO: pip is looking at multiple versions of onnxruntime to determine which version is compatible with other requirements. This could take a while.
INFO: pip is looking at multiple versions of ivadomed to determine which version is compatible with other requirements. This could take a while.
INFO: pip is looking at multiple versions of h5py to determine which version is compatible with other requirements. This could take a while.
INFO: pip is looking at multiple versions of dipy to determine which version is compatible with other requirements. This could take a while.
Collecting dipy
  Downloading dipy-1.4.0-cp37-cp37m-macosx_10_9_x86_64.whl (7.6 MB)
INFO: pip is looking at multiple versions of colored to determine which version is compatible with other requirements. This could take a while.
Collecting colored
  Using cached colored-1.4.2-py3-none-any.whl
INFO: pip is looking at multiple versions of torchvision to determine which version is compatible with other requirements. This could take a while.
INFO: pip is looking at multiple versions of <Python from Requires-Python> to determine which version is compatible with other requirements. This could take a while.
INFO: pip is looking at multiple versions of torch to determine which version is compatible with other requirements. This could take a while.
INFO: pip is looking at multiple versions of pyqt5 to determine which version is compatible with other requirements. This could take a while.
INFO: pip is looking at multiple versions of keras to determine which version is compatible with other requirements. This could take a while.
ERROR: Cannot install -r requirements.txt (line 28) and -r requirements.txt (line 7) because these package versions have conflicting dependencies.

The conflict is caused by:
    ivadomed 2.9.1 depends on tensorboard~=2.4
    tensorflow 1.15.5 depends on tensorboard<1.16.0 and >=1.15.0
    ivadomed 2.9.1 depends on tensorboard~=2.4
    tensorflow 1.15.4 depends on tensorboard<1.16.0 and >=1.15.0
    ivadomed 2.9.1 depends on tensorboard~=2.4
    tensorflow 1.15.3 depends on tensorboard<1.16.0 and >=1.15.0
    ivadomed 2.9.1 depends on tensorboard~=2.4
    tensorflow 1.15.2 depends on tensorboard<1.16.0 and >=1.15.0
    ivadomed 2.9.1 depends on tensorboard~=2.4
    tensorflow 1.15.0 depends on tensorboard<1.16.0 and >=1.15.0

To fix this you could try to:
1. loosen the range of package versions you've specified
2. remove package versions to allow pip attempt to solve the dependency conflict

ERROR: ResolutionImpossible: for help visit https://pip.pypa.io/en/latest/user_guide/#fixing-conflicting-dependencies

Failed running pip install: 1


Installation failed!

Please find the file /Users/julien/code/sct/install_sct_log.txt, 
then upload it as a .txt attachment in a new topic on SCT's forum:
--> http://forum.spinalcordmri.org/c/sct
```

</details>

<details><summary>requirements.txt</summary>

```console
colored
dipy
# h5py is pinned to minor than 3 due to issues with Keras/TF
# https://github.com/tensorflow/tensorflow/issues/44467
h5py~=2.10.0
Keras==2.3.1
git+https://github.com/ivadomed/ivadomed@master
# ivadomed==git+https://github.com/ivadomed/ivadomed/commit/0fdc5a8f7f3a93a86aa94ff3feac3c460c732ff5
matplotlib
nibabel
numpy
# onnxruntime>=1.5.1 requires `brew install libomp` on macOS.
# So, pin to 1.4.0 to avoid having to ask users to install libomp.
# ivadomed==2.5.0 would also do this, but #3035 is preventing that.
# onnxruntime==1.4.0
onnxruntime>=1.7.0
pandas
psutil
pyqt5==5.11.3
pytest
pytest-cov
raven
requests
requirements-parser
scipy
scikit-image
scikit-learn
tensorflow~=1.15.0
# PyTorch's Linux/Windows distribution is very large due to its GPU support,
# but we only need that for training models. For users, use the CPU-only version
# (only available directly from the PyTorch project).
# The macOS version has never had GPU support, so doesn't need the workaround.
-f https://download.pytorch.org/whl/cpu/torch_stable.html
torch==1.5.0+cpu; sys_platform != "darwin"
torch==1.5.0; sys_platform == "darwin"
torchvision==0.6.0+cpu; sys_platform != "darwin"
torchvision==0.6.0; sys_platform == "darwin"
xlwt
tqdm
transforms3d
urllib3[secure]
pytest_console_scripts
wquantiles

```

</details>

@joshuacwnewton as you suggested it is likely related to the tensorboard conflict mentioned in https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/3595#discussion_r782479392

File: requirements.txt
Comment (joshuacwnewton) [inline requirements.txt:7]: I temporarily pointed SCT to my test branch on ivadomed. It looks like the installation went OK! 

Screenshot from 2022-01-18 11-57-54

The reason for the test failures is not a real issue: It's just because `sct_check_dependencies` isn't expecting an installation from `git`. :P

Comment (joshuacwnewton) [reply requirements.txt:7]: I've temporarily addressed this in f7207b2, but it will be undone before this PR is merged.

File: requirements.txt
Comment (joshuacwnewton) [reply requirements.txt]: `ivadomed==2.9.2` has been released, so I'm going to clean up my temporary commits a little, and then these issues should resolve themselves. (Famous last words...)

**Comment by joshuacwnewton** [comment]
Whoops! I've done something silly with Git. Let me... clean that right up. :sweat_smile:

File: requirements.txt
Comment (joshuacwnewton) [reply requirements.txt]: Thank goodness!

Screenshot from 2022-01-18 15-40-32

> `ivadomed-2.9.2`
> `tensorboard-1.15.0 tensorflow-1.15.5`
> `onnxruntime-1.7.0` 

Everything is installed in harmony together with no test failures. :heart:

**Review by RignonNoel — COMMENTED** [review]
From what I read on the issues and on the official source of `onnxruntime` and `libomp` it seems to be good! However I only work with Linux and Windows, so I can't test it for you.

**Comment by joshuacwnewton** [comment]
> From what I read on the issues and on the official source of `onnxruntime` and `libomp` it seems to be good! However I only work with Linux and Windows, so I can't test it for you.

No worries! I think it's alright to lean on our macOS CI runner for testing in this case. :smile: 

And besides, for such a small change, I think I was mainly looking for a quick approval on the PR, since this is needed for https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/3637. (My apologies -- I should have been clearer about what I was looking for when I requested the review. :sweat_smile:)

**Review by RignonNoel — APPROVED** [review]
Here we go! Let's go count on the macOS CI runner for the details ;)

**Pull request merged**
